Scav selling pad

Baldir-Odensen opened the pull request:
## About The Pull Request

adds the selling pad and the computer to control it to the scavengers to be added to the ship.

## Why It's Good For The Game

it would allow the scavengers to sell stuff and thus make money from their scavenging.

## Changelog
:cl:
add: adds a selling pad to for the scavengers along with an accompanying computer
add: adds a budget card for and budget for the scavengers
/:cl:

Baldir-Odensen committed:
Should be it

ike709 commented on code/game/machinery/scav_selling_pad.dm:
```diff
@@ -0,0 +1,205 @@
+	var/cargo_hold_id
+
+/obj/machinery/scav_selling/multitool_act(mob/living/user, obj/item/multitool/I)
+	if (istype(I))
```
Is it even possible for this check to fail?

Baldir-Odensen replied:
no clue it there on the pirate one so i put it there on the scav one

ike709 commented on code/game/machinery/scav_selling_pad.dm:
```diff
@@ -0,0 +1,205 @@
+
+/obj/machinery/scav_selling/multitool_act(mob/living/user, obj/item/multitool/I)
+	if (istype(I))
+		to_chat(user, "<span class='notice'>You register [src] in [I]s buffer.</span>")
```
```suggestion
		to_chat(user, "<span class='notice'>You register \the [src] in [I]'s buffer.</span>")
```

Baldir-Odensen replied:
ok its in the latest commit

ike709 commented on code/game/machinery/scav_selling_pad.dm:
```diff
@@ -0,0 +1,205 @@
+				pad = P
+				return
+	else
+		pad = locate() in range(4,src)
```
Gross.

ike709 commented on code/game/machinery/scav_selling_pad.dm:
```diff
@@ -0,0 +1,205 @@
+		updateDialog()
+		return TRUE
+
+/obj/machinery/computer/scav_selling_control/LateInitialize()
```
IIRC, even using `LateInitialize()`, you can't guarantee that the maploader will load in the other machine first when loading in a shuttle.

Baldir-Odensen replied:
ok so they will have to reconect it once in a while. it is based on the code for the pirate pad so if the pirates dont have that problem this one shouldnt have it either.

ike709 commented on code/game/machinery/scav_selling_pad.dm:
```diff
@@ -0,0 +1,205 @@
+/obj/machinery/computer/scav_selling_control
+	name = "cargo hold control terminal"
+	var/status_report = "Idle"
+	var/obj/machinery/scav_selling/pad
```
This should use weakrefs.

Baldir-Odensen replied:
dont really know how to do those.
im mostly learning this as i work on it

ike709 commented on code/game/machinery/scav_selling_pad.dm:
```diff
@@ -0,0 +1,205 @@
+	else
+		pad = locate() in range(4,src)
+
+/obj/machinery/computer/scav_selling_control/ui_interact(mob/user)
```
I would strongly prefer a tgui-next UI.

Baldir-Odensen replied:
No clue how to set up tgui next.
The pirate selling computer uses the exact same ui code btw

ike709 reviewed (changes requested):
Didn't fully review this, already left you with a fair bit to work on though.

Baldir-Odensen commented:
Don't really know how to do tgui next I mostly used the pirate selling pad and changed a few things

Baldir-Odensen committed:
ok lets remove the ransoming but

removes the abillity to ransom crew and does minor gramatical changes as sugested by ike

Baldir-Odensen committed:
this will not pass travis but i am desperate please send help

please someone tell me where i went wrong with implementing weak refs i have no idea what i am doing wrong and i have to sleep now. the problem is everything regarding the variables 
	icon_state = "lpad-idle-o"
	var/idle_state 	= "lpad-idle-o"
	var/warmup_state = "lpad-idle"
	var/sending_state = "lpad-beam"

qwertyquerty commented:
@Baldir-Odensen ?

qwertyquerty commented:
@Baldir-Odensen

qwertyquerty commented:
stale
